pkg/gds/tokens/tokens_test.go

```diff
@@ -0,0 +1,193 @@
+	// Sign the refresh token
+	rtks, err := tm.Sign(refreshToken)
+	require.NoError(err, "could not sign refresh token")
+	require.NotEqual(atks, rtks, "identical access and refresh tokens")
+
+	// Validate the access token
+	_, err = tm.Verify(atks)
+	require.NoError(err, "could not validate access token")
+
+	// Validate the refresh token (should be invalid because of not before in the future)
+	_, err = tm.Verify(rtks)
+	require.Error(err, "refresh token is valid?")
+}
+
+// TODO: test validation of audience, issuer, and subject.
+// TODO: test time based validation (not before, issued at, expires)
+// TODO: test signed with wrong key
+func (s *TokenTestSuite) TestValidTokens() {
+	require := s.Require()
+	tm, err := tokens.New(s.testdata)
+	require.NoError(err, "could not initialize token manager")
+
+	// Default creds
+	creds := map[string]interface{}{
+		"hd":      "rotational.io",
+		"email":   "kate@rotational.io",
+		"name":    "Kate Holland",
+		"picture": "https://foo.googleusercontent.com/test!/Aoh14gJceTrUA",
+	}
+
+	// Test required creds
+	for _, key := range []string{"hd", "email"} {
+		// remove key
+		orig := creds[key]
+		delete(creds, key)
+
+		// Fail when creating an access token
+		_, err = tm.CreateAccessToken(creds)
+		require.Error(err)
+
+		// replace key
+		creds[key] = orig
+	}
+
+	// Test optional creds
+	for _, key := range []string{"name", "picture"} {
+		// remove key
+		orig := creds[key]
+		delete(creds, key)
+
+		// Fail when creating an access token
+		_, err = tm.CreateAccessToken(creds)
+		require.NoError(err)
+
+		// replace key
+		creds[key] = orig
+	}
+
+	// Test optional and empty
+	creds["name"] = ""
+	_, err = tm.CreateAccessToken(creds)
+	require.NoError(err)
+
+	// Test not optional and empty
+	creds["email"] = ""
+	_, err = tm.CreateAccessToken(creds)
+	require.Error(err)
+
+	// Test bad parsing
+	creds["email"] = 1234
+	_, err = tm.CreateAccessToken(creds)
+	require.Error(err)
+}
+
+// Test that a token signed with an old cert can still be verified.
+func (s *TokenTestSuite) TestKeyRotation() {
+	require := s.Require()
+
+	// Create the "old token manager"
+	testdata := make(map[string]string)
+	testdata["1yAwhf28bXi3IWP6FYcGa0dcrfq"] = "testdata/1yAwhf28bXi3IWP6FYcGa0dcrfq.pem"
+	oldTM, err := tokens.New(testdata)
+	require.NoError(err, "could not initialize old token manager")
+
+	// Create the "new" token manager with the new key
+	newTM, err := tokens.New(s.testdata)
+	require.NoError(err, "could not initialize new token manager")
+
+	// Create a valid token with the "old token manager"
+	token, err := oldTM.CreateAccessToken(map[string]interface{}{
+		"hd":      "rotational.io",
+		"email":   "kate@rotational.io",
+		"name":    "Kate Holland",
+		"picture": "https://foo.googleusercontent.com/test!/Aoh14gJceTrUA",
+	})
+	require.NoError(err)
+
+	tks, err := oldTM.Sign(token)
+	require.NoError(err)
+
+	// Validate token with "new token manager"
+	_, err = newTM.Verify(tks)
+	require.NoError(err)
```

Comment:
Would we want a check for the reverse? e.g. ensure that oldTM.Verify(newTM.CreateAccessToken) returns an error? I suppose this is what you meant by test signed with wrong key in the earlier test